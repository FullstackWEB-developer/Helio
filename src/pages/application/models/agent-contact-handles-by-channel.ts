import {ChannelTypes} from '@shared/models';

export interface AgentContactHandlesByChannel {
    agent: string;
    channel: ChannelTypes;
    startInterval?: Date;
    endInterval?: Date;
    contactsHandled?: number
}
