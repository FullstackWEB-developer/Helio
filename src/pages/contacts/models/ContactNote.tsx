export interface ContactNote {
    id?: number;
    noteText: string;
    createdOn?: Date;
    createdBy?: string;
}