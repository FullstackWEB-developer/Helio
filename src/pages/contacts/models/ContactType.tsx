export enum ContactType {
    Individual = 'Individual',
    Company = 'Company'
}