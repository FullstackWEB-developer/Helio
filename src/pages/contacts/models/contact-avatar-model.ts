import {Icon} from '@components/svg-icon/icon';
export interface ContactAvatarModel {
    initials?: string;
    className?: string; 
    isCompany?: boolean;
    iconType?: Icon;
    iconClassName?: string;
    iconFillClass?: string;
}