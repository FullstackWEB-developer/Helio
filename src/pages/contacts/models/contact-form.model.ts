import {Option} from "@components/option/option";

export interface ContactFormModel {
    id?: string;
    firstName?: string;
    lastName?: string;
    companyName?: string;
    category: number | Option;
    jobTitle?: string;
    department?: string;
    email?: string;
    workMainPhone?: string;
    workMainExtension?: string;
    workDirectPhone?: string;
    mobile?: string;
    fax?: string;
    website?: string;
    primaryAddressLine?: string;
    primaryApt?: string;
    primaryCity?: string;
    primaryState?: string | Option;
    primaryZipCode?: string;
    shippingAddressLine?: string;
    shippingApt?: string;
    shippingCity?: string;
    shippingState?: string | Option;
    shippingZipCode?: string;
    billingAddressLine?: string;
    billingApt?: string;
    billingCity?: string;
    billingState?: string | Option;
    billingZipCode?: string;
    isStarred?: boolean;
    relatedId?: string;
}