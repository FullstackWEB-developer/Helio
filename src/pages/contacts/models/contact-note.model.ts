export interface ContactNote {
    id?: string;
    noteText: string;
    createdOn: Date;
    createdByName: string;
    contactId: string;
}
export interface AddContactNoteProps {
    contactId: string;
    contactNoteDto: ContactNote
}