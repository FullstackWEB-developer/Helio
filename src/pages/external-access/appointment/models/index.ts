export * from './appointment-type.model';
export * from './appointment-slot.model';
export * from './appointment.model';
export * from './appointment-cancellation.model';
export * from './appointment-cancel-reason.model';
export * from './appointment-slot-request.model';
