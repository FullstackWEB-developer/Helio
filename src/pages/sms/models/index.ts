export * from "./sms-new-message-steps";
export * from "./sms-notification-data";
export * from "./sms-query-type";
