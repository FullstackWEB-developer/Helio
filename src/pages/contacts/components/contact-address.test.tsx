import i18n from '../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {act, render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import ContactAddress from './contact-address';
import { AddressType } from '@shared/models';
import { useForm } from 'react-hook-form';
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("ContactAddress tests", () => {
    let container: HTMLDivElement | null;
    let mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {

        },
        ticketState: {
            lookupValues: []
        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        },
        ccpState: {
            chatCounter: 1,
            voiceCounter: 2
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders ContactAddress-PrimaryAddress correctly", async () => {
        const Component = () => {
            const { control } = useForm<{
                test: string;
                test1: { test: string }[];
            }>();

            return <TestWrapper mockState={mockState}>
                <ContactAddress title='Test' addressType={AddressType.PrimaryAddress} control={control} />
            </TestWrapper>
        };
        let fragment: any;
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <Component/>
            </TestWrapper>);
            fragment = asFragment;
        })
        expect(fragment()).toMatchSnapshot();
        
    });

    it("renders ContactAddress-BillingAddress correctly", async () => {
        const Component = () => {
            const { control } = useForm<{
                test: string;
                test1: { test: string }[];
            }>();

            return <TestWrapper mockState={mockState}>
                <ContactAddress title='Test' addressType={AddressType.BillingAddress} control={control} />
            </TestWrapper>
        };
        let fragment: any;
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <Component/>
            </TestWrapper>);
            fragment = asFragment;
        })
        expect(fragment()).toMatchSnapshot();
        
    });

    it("renders ContactAddress-ShippingAddress correctly", async () => {
        const Component = () => {
            const { control } = useForm<{
              test: string;
              test1: { test: string }[];
            }>();
      
            return <TestWrapper mockState={mockState}>
                <ContactAddress title='Test' addressType={AddressType.ShippingAddress} control={control} />
            </TestWrapper>
          };
        let fragment: any;
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <Component/>
            </TestWrapper>);
            fragment = asFragment;
        })
        expect(fragment()).toMatchSnapshot();
        
    });
})
