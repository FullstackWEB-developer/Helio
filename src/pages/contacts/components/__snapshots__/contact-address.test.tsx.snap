// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactAddress tests renders ContactAddress-BillingAddress correctly 1`] = `
<DocumentFragment>
  <div
    class="grid grid-cols-1 lg:grid-cols-12 gap-x-8 h-10 mt-2 mb-4 items-center"
  >
    <div
      class="col-span-12 lg:col-span-5"
    >
      Test
    </div>
    <div
      class="col-span-12 lg:col-span-3"
    />
    <div
      class="col-span-12 lg:col-span-2"
    >
      <div
        class="flex items-center cursor-pointer"
      >
        <div
          class="pl-4 pr-2"
        >
          <div
            class=""
          >
            <svg
              class="icon-medium "
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                transform="translate(-889 -16)"
              >
                <rect
                  fill="none"
                  height="24"
                  transform="translate(889 16)"
                  width="24"
                />
                <path
                  class="contact-light-fill"
                  d="M12,2A10,10,0,1,0,22,12,9.991,9.991,0,0,0,12,2Zm5,13.59L15.59,17,12,13.41,8.41,17,7,15.59,10.59,12,7,8.41,8.41,7,12,10.59,15.59,7,17,8.41,13.41,12Z"
                  transform="translate(889 16)"
                />
              </g>
            </svg>
          </div>
        </div>
        <span
          class="body2"
        >
          common.remove
        </span>
      </div>
    </div>
  </div>
  <div
    class="relative"
  >
    <div
      class="h-9"
    >
      <label
        class="flex flex-row items-center checkbox-button"
      >
        <input
          class="checkbox"
          id="_"
          name=""
          type="checkbox"
        />
        <span
          class="checkbox-control"
        >
          <div
            class=""
          >
            <svg
              class="icon-medium "
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="24"
                width="24"
              />
              <path
                class="svg-checkbox"
                d="M16,18H2a2,2,0,0,1-2-2V2A2,2,0,0,1,2,0H16a2,2,0,0,1,2,2V16A2,2,0,0,1,16,18ZM3.41,7.59h0L2,9l5,5,9-9L14.59,3.58,7,11.17,3.41,7.59Z"
              />
            </svg>
          </div>
        </span>
        <div
          class="truncate relative w-full h-full inline-block"
        >
          <span
            class="w-60 w-60"
          >
            contacts.contact_details.individual.same_as_primary
          </span>
        </div>
      </label>
    </div>
  </div>
  <div
    class="grid grid-cols-1 lg:grid-cols-12 gap-x-8"
  >
    <div
      class="col-span-12 lg:col-span-5"
    >
      <div
        class="flex flex-col h-20 input-group"
      >
        <div
          class="input-group-container flex flex-wrap items-stretch w-full relative  "
        >
          <input
            autocomplete="off"
            class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
            data-test-id="contact-billing-address-line"
            data-testid="billingAddressLine"
            name="billingAddressLine"
            placeholder=""
            type="text"
            value=""
          />
          <label
            class="absolute truncate  body2 "
          >
            contacts.contact_details.individual.address
          </label>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 lg:col-span-5"
    >
      <div
        class="flex flex-col h-20 input-group"
      >
        <div
          class="input-group-container flex flex-wrap items-stretch w-full relative  "
        >
          <input
            autocomplete="off"
            class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
            data-test-id="contactbilling-address-apt"
            data-testid="billingApt"
            name="billingApt"
            placeholder=""
            type="text"
            value=""
          />
          <label
            class="absolute truncate  body2 "
          >
            contacts.contact_details.individual.apt
          </label>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 lg:col-span-5"
    >
      <div
        class="flex flex-col h-20 input-group"
      >
        <div
          class="input-group-container flex flex-wrap items-stretch w-full relative  "
        >
          <input
            autocomplete="off"
            class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
            data-test-id="contact-billing-address-city"
            data-testid="billingCity"
            name="billingCity"
            placeholder=""
            type="text"
            value=""
          />
          <label
            class="absolute truncate  body2 "
          >
            contacts.contact_details.individual.city
          </label>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 lg:col-span-3"
    >
      <div
        class="select-wrapper relative w-full flex flex-col h-20 "
      >
        <div
          class="select relative flex flex-col "
        >
          <input
            autocomplete="nope"
            autocorrect="off"
            class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger   "
            data-testid="billingState"
            name="billingState"
            readonly=""
            spellcheck="false"
            type="text"
            value=""
          />
          <label
            class="select-label absolute truncate"
          >
            <span
              class="select-label-span body2"
            >
              contacts.contact_details.individual.state
            </span>
          </label>
          <div
            class="absolute pt-4 right-4 "
          >
            <div
              class="flex flex-row items-start"
            >
              <div
                class="cursor-pointer"
              >
                <div
                  class=""
                >
                  <svg
                    class="icon-medium "
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="24"
                      width="24"
                    />
                    <path
                      class="select-arrow-fill"
                      d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                      transform="translate(5 16.273) rotate(-90)"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="absolute py-2 options"
          >
            <div
              class="w-full pt-2 text-center subtitle3-small"
            >
              common.autocomplete_search
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 lg:col-span-2"
    >
      <div
        class="flex flex-col h-20 input-group"
      >
        <div
          class="input-group-container flex flex-wrap items-stretch w-full relative  "
        >
          <input
            autocomplete="off"
            class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
            data-test-id="contact-billing-zip-code"
            data-testid="billingZipCode"
            maxlength="5"
            name="billingZipCode"
            placeholder=""
            type="text"
            value=""
          />
          <label
            class="absolute truncate  body2 "
          >
            contacts.contact_details.individual.zip_code
          </label>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ContactAddress tests renders ContactAddress-PrimaryAddress correctly 1`] = `
<DocumentFragment>
  <div
    class="h-10 flex items-center mt-4 mb-2"
  >
    Test
  </div>
  <div
    class="grid grid-cols-1 lg:grid-cols-12 gap-x-8"
  >
    <div
      class="col-span-12 lg:col-span-5"
    >
      <div
        class="flex flex-col h-20 input-group"
      >
        <div
          class="input-group-container flex flex-wrap items-stretch w-full relative  "
        >
          <input
            autocomplete="off"
            class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
            data-test-id="contact-primary-address-line"
            data-testid="primaryAddressLine"
            name="primaryAddressLine"
            placeholder=""
            type="text"
            value=""
          />
          <label
            class="absolute truncate  body2 "
          >
            contacts.contact_details.individual.address
          </label>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 lg:col-span-5"
    >
      <div
        class="flex flex-col h-20 input-group"
      >
        <div
          class="input-group-container flex flex-wrap items-stretch w-full relative  "
        >
          <input
            autocomplete="off"
            class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
            data-test-id="contactprimary-address-apt"
            data-testid="primaryApt"
            name="primaryApt"
            placeholder=""
            type="text"
            value=""
          />
          <label
            class="absolute truncate  body2 "
          >
            contacts.contact_details.individual.apt
          </label>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 lg:col-span-5"
    >
      <div
        class="flex flex-col h-20 input-group"
      >
        <div
          class="input-group-container flex flex-wrap items-stretch w-full relative  "
        >
          <input
            autocomplete="off"
            class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
            data-test-id="contact-primary-address-city"
            data-testid="primaryCity"
            name="primaryCity"
            placeholder=""
            type="text"
            value=""
          />
          <label
            class="absolute truncate  body2 "
          >
            contacts.contact_details.individual.city
          </label>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 lg:col-span-3"
    >
      <div
        class="select-wrapper relative w-full flex flex-col h-20 "
      >
        <div
          class="select relative flex flex-col "
        >
          <input
            autocomplete="nope"
            autocorrect="off"
            class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger   "
            data-testid="primaryState"
            name="primaryState"
            readonly=""
            spellcheck="false"
            type="text"
            value=""
          />
          <label
            class="select-label absolute truncate"
          >
            <span
              class="select-label-span body2"
            >
              contacts.contact_details.individual.state
            </span>
          </label>
          <div
            class="absolute pt-4 right-4 "
          >
            <div
              class="flex flex-row items-start"
            >
              <div
                class="cursor-pointer"
              >
                <div
                  class=""
                >
                  <svg
                    class="icon-medium "
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="24"
                      width="24"
                    />
                    <path
                      class="select-arrow-fill"
                      d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                      transform="translate(5 16.273) rotate(-90)"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="absolute py-2 options"
          >
            <div
              class="w-full pt-2 text-center subtitle3-small"
            >
              common.autocomplete_search
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 lg:col-span-2"
    >
      <div
        class="flex flex-col h-20 input-group"
      >
        <div
          class="input-group-container flex flex-wrap items-stretch w-full relative  "
        >
          <input
            autocomplete="off"
            class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
            data-test-id="contact-primary-zip-code"
            data-testid="primaryZipCode"
            maxlength="5"
            name="primaryZipCode"
            placeholder=""
            type="text"
            value=""
          />
          <label
            class="absolute truncate  body2 "
          >
            contacts.contact_details.individual.zip_code
          </label>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ContactAddress tests renders ContactAddress-ShippingAddress correctly 1`] = `
<DocumentFragment>
  <div
    class="grid grid-cols-1 lg:grid-cols-12 gap-x-8 h-10 mt-2 mb-4 items-center"
  >
    <div
      class="col-span-12 lg:col-span-5"
    >
      Test
    </div>
    <div
      class="col-span-12 lg:col-span-3"
    />
    <div
      class="col-span-12 lg:col-span-2"
    >
      <div
        class="flex items-center cursor-pointer"
      >
        <div
          class="pl-4 pr-2"
        >
          <div
            class=""
          >
            <svg
              class="icon-medium "
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                transform="translate(-889 -16)"
              >
                <rect
                  fill="none"
                  height="24"
                  transform="translate(889 16)"
                  width="24"
                />
                <path
                  class="contact-light-fill"
                  d="M12,2A10,10,0,1,0,22,12,9.991,9.991,0,0,0,12,2Zm5,13.59L15.59,17,12,13.41,8.41,17,7,15.59,10.59,12,7,8.41,8.41,7,12,10.59,15.59,7,17,8.41,13.41,12Z"
                  transform="translate(889 16)"
                />
              </g>
            </svg>
          </div>
        </div>
        <span
          class="body2"
        >
          common.remove
        </span>
      </div>
    </div>
  </div>
  <div
    class="relative"
  >
    <div
      class="h-9"
    >
      <label
        class="flex flex-row items-center checkbox-button"
      >
        <input
          class="checkbox"
          id="_"
          name=""
          type="checkbox"
        />
        <span
          class="checkbox-control"
        >
          <div
            class=""
          >
            <svg
              class="icon-medium "
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="24"
                width="24"
              />
              <path
                class="svg-checkbox"
                d="M16,18H2a2,2,0,0,1-2-2V2A2,2,0,0,1,2,0H16a2,2,0,0,1,2,2V16A2,2,0,0,1,16,18ZM3.41,7.59h0L2,9l5,5,9-9L14.59,3.58,7,11.17,3.41,7.59Z"
              />
            </svg>
          </div>
        </span>
        <div
          class="truncate relative w-full h-full inline-block"
        >
          <span
            class="w-60 w-60"
          >
            contacts.contact_details.individual.same_as_primary
          </span>
        </div>
      </label>
    </div>
  </div>
  <div
    class="grid grid-cols-1 lg:grid-cols-12 gap-x-8"
  >
    <div
      class="col-span-12 lg:col-span-5"
    >
      <div
        class="flex flex-col h-20 input-group"
      >
        <div
          class="input-group-container flex flex-wrap items-stretch w-full relative  "
        >
          <input
            autocomplete="off"
            class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
            data-test-id="contact-shipping-address-line"
            data-testid="shippingAddressLine"
            name="shippingAddressLine"
            placeholder=""
            type="text"
            value=""
          />
          <label
            class="absolute truncate  body2 "
          >
            contacts.contact_details.individual.address
          </label>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 lg:col-span-5"
    >
      <div
        class="flex flex-col h-20 input-group"
      >
        <div
          class="input-group-container flex flex-wrap items-stretch w-full relative  "
        >
          <input
            autocomplete="off"
            class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
            data-test-id="contactshipping-address-apt"
            data-testid="shippingApt"
            name="shippingApt"
            placeholder=""
            type="text"
            value=""
          />
          <label
            class="absolute truncate  body2 "
          >
            contacts.contact_details.individual.apt
          </label>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 lg:col-span-5"
    >
      <div
        class="flex flex-col h-20 input-group"
      >
        <div
          class="input-group-container flex flex-wrap items-stretch w-full relative  "
        >
          <input
            autocomplete="off"
            class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
            data-test-id="contact-shipping-address-city"
            data-testid="shippingCity"
            name="shippingCity"
            placeholder=""
            type="text"
            value=""
          />
          <label
            class="absolute truncate  body2 "
          >
            contacts.contact_details.individual.city
          </label>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 lg:col-span-3"
    >
      <div
        class="select-wrapper relative w-full flex flex-col h-20 "
      >
        <div
          class="select relative flex flex-col "
        >
          <input
            autocomplete="nope"
            autocorrect="off"
            class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger   "
            data-testid="shippingState"
            name="shippingState"
            readonly=""
            spellcheck="false"
            type="text"
            value=""
          />
          <label
            class="select-label absolute truncate"
          >
            <span
              class="select-label-span body2"
            >
              contacts.contact_details.individual.state
            </span>
          </label>
          <div
            class="absolute pt-4 right-4 "
          >
            <div
              class="flex flex-row items-start"
            >
              <div
                class="cursor-pointer"
              >
                <div
                  class=""
                >
                  <svg
                    class="icon-medium "
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="24"
                      width="24"
                    />
                    <path
                      class="select-arrow-fill"
                      d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                      transform="translate(5 16.273) rotate(-90)"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="absolute py-2 options"
          >
            <div
              class="w-full pt-2 text-center subtitle3-small"
            >
              common.autocomplete_search
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 lg:col-span-2"
    >
      <div
        class="flex flex-col h-20 input-group"
      >
        <div
          class="input-group-container flex flex-wrap items-stretch w-full relative  "
        >
          <input
            autocomplete="off"
            class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
            data-test-id="contact-shipping-zip-code"
            data-testid="shippingZipCode"
            maxlength="5"
            name="shippingZipCode"
            placeholder=""
            type="text"
            value=""
          />
          <label
            class="absolute truncate  body2 "
          >
            contacts.contact_details.individual.zip_code
          </label>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
